pytket/pytket/wasm/wasm.py

```diff
@@ -34,6 +36,107 @@ def __init__(self, filepath: str):
 
         self._wasmuid = hashlib.md5(self._wasm_file_encoded).hexdigest()
 
+        wasm_module = Module(Store(), open(self._filepath, "rb").read())
+        instance = Instance(wasm_module)
+
+        # stores the names of the functions mapped
+        #  to the number of parameters and the number of return values
+        self._functions = dict()
+
+        # contains the list of functions that are not allowed
+        # to use in pytket (because of types that are not i32)
+        self._unvalid_functions = []
+
+        self._wasm_file = base64.decodebytes(self._wasm_file_encoded)
+
+        for wasm_obj in instance.exports:
+            if len(wasm_obj) > 1 and isinstance(wasm_obj[1], Function):
+                valid_function = True
+                wasm_function = wasm_obj[1]
+
+                # the direct evaluation of the types converts to python
+                #  ints which remove the information if we are working
+                #  with i32 or i64, so we are unfortunately required
+                #  with the str version of the signature in the wasm file
```

Comment:
Not quite sure what the problem is here. Would using `numpy.int32` and friends (instead of python integers) help?

Reply:
The problem is in the wasmer module, I would assume they will fix that with using numpy or a new own type in the future.

Reply:
I have not checked the bug tracked of them yet, will do that later...